jax/_src/errors.py

```diff
@@ -462,6 +408,117 @@ def __init__(self, tracer: core.Tracer):
         f"The __index__() method was called on the JAX Tracer object {tracer}")
 
 
+@export
+class TracerBoolConversionError(ConcretizationTypeError):
+  """
+  This error occurs when a traced value in JAX is used in a context where a
+  boolean value is expected (see :ref:`faq-different-kinds-of-jax-values`
+  for more on what a Tracer is).
+
+  The boolean cast may be an explicit (e.g. ``bool(x)``) or implicit, through use of
+  control flow (e.g. ``if x > 0:``) or use of Python comparison operators
+  (e.g. ``z = x and y`` or ``z = min(x, y)``).
```

Comment:
I'd also add the typical Python idiom of:
```python
z = nullable_value or default_value
```
Where optional value may be `None`. That's how I encountered this issue. The difference is it might not be obvious to a distracted programmer that `or` is a boolean operation here, since there aren't any booleans on either side of `or`.

Reply:
Thanks, done!